### PR TITLE
Integrate tcib in reproducer

### PR DESCRIPTION
**Pull request by frenzyfriday**
As a pull request owner and reviewers, I checked that:
- [X] Appropriate testing is done and actually running
- [X] Appropriate documentation exists and/or is up-to-date:
  - [X] README in the role
  - [X] Content of the docs/source is reflecting the changes

### CODE REVIEW

File: roles/reproducer/tasks/ci_job.yml [hunk 1]
```diff
@@ -51,6 +51,15 @@
             dest: "/home/zuul/src/github.com/openstack-k8s-operators/ci-framework/{{ job_id }}_content-provider.yml"
             src: "content-provider.yml.j2"
 
+        - name: Push tcib playbook if needed
+          tags:
+            - bootstrap
+          when:
+            - use_tcib | default(false) | bool
```
Comment (raukadah) [inline roles/reproducer/tasks/ci_job.yml:58]:
```suggestion
            - use_openstack_content_provider | default(false) | bool
```

**Comment by raukadah** [comment]
@frenzyfriday Thank you for adding support for tcib content provider. Since we are changing the var name. This part https://github.com/openstack-k8s-operators/ci-framework/blob/ad675ad39fcec888c70bbe7dc6d33a637a8ac2eb/roles/reproducer/tasks/ci_data.yml#L63 needs to be fixed with new var names.

File: roles/reproducer/templates/play.yml.j2 [hunk 2]
```diff
@@ -6,7 +6,7 @@
       - '@~/{{ job_id }}-params/custom-params.yml'
       - '@~/{{ job_id }}-params/install-yamls-params.yml'
       - '@~/{{ job_id }}-params/reproducer_params.yml'
-{% if use_content_provider | bool %}
+{% if ((operator_content_provider) | bool) or ((openstack_content_provider) | bool)) %}
```
Comment (raukadah) [inline roles/reproducer/templates/play.yml.j2:9]:
```suggestion
{% if (operator_content_provider | bool) or (openstack_content_provider | bool) %}
```

File: roles/reproducer/templates/content-provider.yml.j2 [hunk 3]
```diff
@@ -54,54 +54,142 @@
...
+                - cifmw_zuul_target_host != 'all'
+                - inventory_hostname != cifmw_zuul_target_host
+              ansible.builtin.meta: end_host
+
+        - name: Discover the host ip
+          ansible.builtin.set_fact:
+            node_ip: >-
+              {{ hostvars[groups.all[0]].ansible_host if hostvars[groups.all[0]].ansible_host
+              is match("[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+")
+              else hostvars[inventory_hostname]['ansible_default_ipv4']['address'] }}
```
Comment (raukadah) [inline roles/reproducer/templates/content-provider.yml.j2:136]: This part is not needed. We can replace node_ip to cifmw_rp_registry_ip. or we can modify this task.
```suggestion
        - name: Discover the host ip
          ansible.builtin.set_fact:
            node_ip: "{{ cifmw_rp_registry_ip }} "
```

File: roles/reproducer/templates/content-provider.yml.j2 [hunk 4]
```diff
@@ -54,54 +54,138 @@
...
+              {{ _dlrn_md5 | default('nowhere') }}
+    {% endraw %}
+
+        - name: Set up content registry IP address
+          zuul_return:
+            data:
+              zuul:
+                pause: true
+              content_provider_registry_ip: "{{ node_ip | default('nowhere') }}"
+              content_provider_dlrn_md5_hash: "{{ _dlrn_md5 | default('nowhere') }}"
```
Comment (raukadah) [inline roles/reproducer/templates/content-provider.yml.j2:191]: We donot need this part. Please drop it.

File: roles/reproducer/templates/content-provider.yml.j2 [hunk 5]
```diff
@@ -54,54 +54,138 @@
...
+                _dlrn_md5: "{{ _md5_data['content'] | b64decode }}"
+              when: _check_dlrn_md5.stat.exists
+
+        - name: Return Zuul Data
+          ansible.builtin.debug:
+            msg: >-
+              Running Content provider registry on
+              {{ node_ip | default('nowhere') }} with dlrn md5 hash
+              {{ _dlrn_md5 | default('nowhere') }}
+    {% endraw %}
```
Comment (raukadah) [inline roles/reproducer/templates/content-provider.yml.j2:183]:
```suggestion
{% endraw %}
```

File: roles/reproducer/templates/content-provider.yml.j2 (hunk 4 above)
Comment (frenzyfriday) [reply roles/reproducer/templates/content-provider.yml.j2:191]: ack, removed

File: roles/reproducer/templates/content-provider.yml.j2 (hunk 3 above)
Comment (frenzyfriday) [reply roles/reproducer/templates/content-provider.yml.j2]: ack, done

File: roles/reproducer/templates/play.yml.j2 (hunk 2 above)
Comment (frenzyfriday) [reply roles/reproducer/templates/play.yml.j2]: ack, done

**Comment by cjeanner** [comment]
recheck

File: roles/reproducer/tasks/ci_data.yml [hunk 6]
```diff
@@ -60,7 +60,11 @@
       ansible.builtin.set_fact:
         cacheable: true
         ci_job_networking: "{{ zuul_inventory.all.hosts.crc.crc_ci_bootstrap_networking }}"
-        use_content_provider: >-
```
Comment (cjeanner) [inline roles/reproducer/tasks/ci_data.yml:63]: I don't get it - why changing the name, and having 2 facts with the exact same value?

Comment (viroel) [reply roles/reproducer/tasks/ci_data.yml:63]: I still don't get this too.
if `zuul_params.content_provider_registry_ip`  is defined, both `openstack_content_provider` and `operator_content_provider` will be set to `true`

Comment (frenzyfriday) [reply roles/reproducer/tasks/ci_data.yml:63]: yeah, sorry. Updating this in a few mins

Comment (cjeanner) [reply roles/reproducer/tasks/ci_data.yml:63]: ah, now I get it :)

File: roles/reproducer/templates/content-provider.yml.j2 [hunk 7]
```diff
@@ -105,3 +106,23 @@
...
+      ansible.builtin.command:
+        chdir: "/home/zuul/src/github.com/openstack-k8s-operators/ci-framework"
+        cmd: >-
+          ansible-playbook ci/playbooks/tcib/tcib.yml
+          -i "~/ci-framework-data/artifacts/zuul_inventory.yml"
+          -e @./scenarios/centos-9/base.yml
+          -e @./scenarios/centos-9/tcib.yml
+          -e cifmw_zuul_target_host=controller-0
+          -e "cifmw_rp_registry_ip="{{ hostvars['localhost']['cifmw_rp_registry_ip'] }}""
+          -e "@{{ ansible_user_dir }}/ci-framework-data/artifacts/parameters/zuul-params.yml"
```
Comment (raukadah) [inline roles/reproducer/templates/content-provider.yml.j2:125]:
```suggestion
         -e "@{{ ansible_user_dir }}/reproducer-variables.yml"
         -e "@{{ ansible_user_dir }}/ci-framework-data/artifacts/parameters/zuul-params.yml"
```

File: roles/reproducer/templates/content-provider.yml.j2 [hunk 8]
```diff
@@ -105,3 +106,23 @@
...
+          --skip-tags build_openstack_packages
```
Comment (raukadah) [inline roles/reproducer/templates/content-provider.yml.j2:126]:
```suggestion
          --skip-tags build_openstack_packages
          
     - name: Output needed content into consumable environment
      ansible.builtin.copy:
        dest: "{{ ansible_user_dir }}/{{ job_id }}-params/content-provider.yml"
        content: |-
          {{
            {'content_provider_registry_ip': cifmw_rp_registry_ip
            } | to_nice_yaml
          }}
          
          
```

**Review by raukadah — COMMENTED** [review]
@frenzyfriday Two more improvements then we are good to go.
- reproducer-variables.yml to pass the all vars passed via reproducer.yml playbook
- {{ ansible_user_dir }}/{{ job_id }}-params/content-provider.yml file will be passed to the reproducer job so that they can consume the containers from controller registry.

**Comment by raukadah** [comment]
/approve
tcib is working fine locally. container build script failed due to missing repos in one of the tobiko container.
```
2024-03-28 03:23:16,254 p=35164 u=zuul n=ansible | fatal: [localhost]: FAILED! => {"changed": true, "cmd": "set -o pipefail && bash build_containers.sh 2>&1 | awk '{ print strftime(\"%Y-%m-%d %H:%M:%S |\"), $0; fflush(); }' > /home/zuul/ci-framework-data/logs/build.log\n", "delta": "0:41:14.298620", "end": "2024-03-28 03:23:16.219979", "msg": "non-zero return code", "rc": 1, "start": "2024-03-28 02:42:01.921359", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}
2024-03-28 03:23:16,256 p=35164 u=zuul n=ansible | NO MORE HOSTS LEFT *************************************************************
2024-03-28 03:23:16,257 p=35164 u=zuul n=ansible | PLAY RECAP *********************************************************************
2024-03-28 03:23:16,257 p=35164 u=zuul n=ansible | localhost                  : ok=54   changed=14   unreachable=0    failed=1    skipped=15   rescued=1    ignored=0   
2024-03-28 03:23:16,258 p=35164 u=zuul n=ansible | Thursday 28 March 2024  03:23:16 -0400 (0:41:14.549)       0:48:23.766 ******** 
2024-03-28 03:23:16,258 p=35164 u=zuul n=ansible | =============================================================================== 
2024-03-28 03:23:16,258 p=35164 u=zuul n=ansible | build_containers : Run container_build.sh script --------------------- 2474.55s
2024-03-28 03:23:16,258 p=35164 u=zuul n=ansible | ci_setup : Install needed packages ------------------------------------ 255.46s
2024-03-28 03:23:16,258 p=35164 u=zuul n=ansible | repo_setup : Generate repos using rhos-release ceph-7.0-rhel-9 -r 9.3 - 103.52s
2024-03-28 03:23:16,258 p=35164 u=zuul n=ansible | registry_deploy : Install Podman package -------------------------------- 9.56s
2024-03-28 03:23:16,259 p=35164 u=zuul n=ansible | repo_setup : Run repo-setup --------------------------------------------- 8.02s
2024-03-28 03:23:16,259 p=35164 u=zuul n=ansible | repo_setup : Install RHOS Release tool ---------------------------------- 7.72s
2024-03-28 03:23:16,259 p=35164 u=zuul n=ansible | repo_setup : Get repo-setup repository ---------------------------------- 6.65s
2024-03-28 03:23:16,259 p=35164 u=zuul n=ansible | discover_latest_image : Get latest image -------------------------------- 4.56s
2024-03-28 03:23:16,259 p=35164 u=zuul n=ansible | install_ca : Install internal CA from url ------------------------------- 4.34s
2024-03-28 03:23:16,259 p=35164 u=zuul n=ansible | ci_setup : Check rhsm status -------------------------------------------- 4.29s
2024-03-28 03:23:16,259 p=35164 u=zuul n=ansible | repo_setup : Run repo-setup-get-hash ------------------------------------ 4.10s
2024-03-28 03:23:16,259 p=35164 u=zuul n=ansible | repo_setup : Initialize python venv and install requirements ------------ 2.17s
2024-03-28 03:23:16,259 p=35164 u=zuul n=ansible | repo_setup : Make sure git-core package is installed -------------------- 1.28s
2024-03-28 03:23:16,259 p=35164 u=zuul n=ansible | repo_setup : Remove existing repos from /etc/yum.repos.d directory ------ 1.15s
2024-03-28 03:23:16,259 p=35164 u=zuul n=ansible | install_ca : Update ca bundle ------------------------------------------- 1.14s
2024-03-28 03:23:16,260 p=35164 u=zuul n=ansible | build_containers : Install required packages ---------------------------- 1.11s
2024-03-28 03:23:16,260 p=35164 u=zuul n=ansible | ci_setup : Manage directories ------------------------------------------- 1.06s
2024-03-28 03:23:16,260 p=35164 u=zuul n=ansible | build_containers : Install tcib from RPM -------------------------------- 1.05s
2024-03-28 03:23:16,260 p=35164 u=zuul n=ansible | install_yamls : Get environment structure ------------------------------- 0.82s
2024-03-28 03:23:16,260 p=35164 u=zuul n=ansible | registry_deploy : Deploy the local registry from docker.io/library/registry:2 --- 0.71s

```

**Pull request merged**